### ISSUE TITLE
Support of GQA-like broadcast within LX resident tensors

### ISSUE THREAD

**Issue by bringlein** [torch-spyre/torch-spyre#4123]
### What feature would you like to be added?

Follow-on to issue https://github.com/torch-spyre/torch-spyre/issues/4055 / PR https://github.com/torch-spyre/torch-spyre/pull/4061 makes a gathered KV page LX-resident when the value matmul has a SINGLE leading batch dim (the page is a clean 1->all-cores broadcast). It does NOT cover the grouped-query-attention (GQA) case, which is what the real paged-attention kernel emits: the value matmul carries TWO batch dims [kv_head, queries_per_kv] and the page is broadcast over the queries_per_kv axis.

In that case the page is physically partitioned across all cores along its kv_head axis, but each consumer core's view of the page is indexed by a DIFFERENT axis (the head-group), so the page ends up owned by fewer cores than the matmul runs on. Bridging that requires a cross-dim (transpose-like) regroup, which the
current grouped-gather / grouped-broadcast classifier rejects with "grouped destination does not evenly contract the source".

Expected today (PR #4061), with the LX-relayout path enabled:

```
    LAYOUT_SOLVER=greedy SPYRE_LX_PLANNER_RELAYOUT=1 \
      SPYRE_INDUCTOR_LOG=1 SPYRE_INDUCTOR_LOG_LEVEL=DEBUG \
      python reproducer_gqa.py 2>&1 | tee /tmp/out.txt

    # the gathered page stays in HBM:
    grep "core div mismatch: broadcast read" /tmp/out.txt
    grep "grouped destination does not evenly contract the source" /tmp/out.txt
    # and no LX allocation is emitted for the page:
    grep -c "allocation={'lx'" /tmp/out.txt        # 0 for the page path
```

Desired once a cross-dim regroup is supported: the page clone is LX-resident
(lx_pinning: ... (clone) -> lx) and an accepted grouped relayout appears.

Minimal reproducer:
```python
import torch

NUM_KV_HEADS = 8       # KV heads
QUERIES_PER_KV = 4     # GQA group size (>1 is what breaks; ==1 is MHA and classifies fine)
LQ = 512               # padded query length
BLK = 128              # page/block size
D = 128                # head size
NUM_PAGES = 64


def kernel(probs, v_pages, page_index_table):
    # One block of the online-softmax loop's V path, GQA shape.
    # probs:   [KV, QPK, LQ, BLK]     (two batch dims)
    # v_pages: [NUM_PAGES, KV, BLK, D]  head-major page pool (no on-device permute)
    page_idx = page_index_table[0, 0:1]
    v_page = v_pages.index_select(0, page_idx)          # [1, KV, BLK, D] indirect gather
    v_page = v_page.squeeze(0).unsqueeze(1)             # [KV, 1, BLK, D] broadcast over QPK
    return torch.matmul(probs, v_page)                  # [KV, QPK, LQ, D]


def main():
    dev = torch.device("spyre")
    probs = torch.randn(NUM_KV_HEADS, QUERIES_PER_KV, LQ, BLK, dtype=torch.float16).to(dev)
    v_pages = torch.randn(NUM_PAGES, NUM_KV_HEADS, BLK, D, dtype=torch.float16).to(dev)
    page_index_table = torch.zeros(NUM_PAGES, 32, dtype=torch.int32).to(dev)

    compiled = torch.compile(kernel, dynamic=False)
    out = compiled(probs, v_pages, page_index_table)
    print("out", tuple(out.shape))


if __name__ == "__main__":
    main()
```

### Why is this feature needed?

For the decode performance of paged-attention, it is essential we do not create unnecessary traffic on the memory-bus. Hence, the (indirectly) gathered KV pages should be LX resident, even for GQA models.

**Comment by AdnanHoque** [comment]
I captured the actual source and consumer views on the #4061 stack. This is not the existing grouped-broadcast case:

- source: 32 unique slices along physical dimension 1;
- destination: 2 unique slices along physical dimension 2, each repeated on 16 cores;
- source span: 1,016,832 bytes per core; destination span: 1,048,448 bytes per core.

A temporary generic classifier reached normal paired allocation, but the two required LX buffers total 2,096,896 bytes per core. At the default `DXP_LX_FRAC_AVAIL=0.2`, the available planner budget is 1,587 KB, so the group correctly falls back to HBM and emits no shuffle.

This therefore should not be folded into #4061 as a classifier-only extension. It needs a separate design for a compact destination layout or a proven safe in-place transfer; otherwise accepting the geometry would only create a relayout that cannot materialize. #4061 remains the correct fix for the one-axis broadcast in #4055.

**Comment by AdnanHoque** [comment]
Correction to my earlier comment: this does fit once we store only the small V piece each core needs.

`32 V fragments → 16 complete V slices → each slice used by 2 query cores`

#4152 proves this on device with one LX move.
